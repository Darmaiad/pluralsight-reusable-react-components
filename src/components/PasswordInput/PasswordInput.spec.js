import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import PasswordInput from './PasswordInput';

// Snapshot test
//  -  Snapshot tests will always pass the first time they run, since they take a snapshot of the component output.

test('Hides password quality bar by default', () => {
    const tree = renderer.create(<PasswordInput
        htmlId="test"
        name="test"
        onChange={() => { }}
        value="Uisi38#8iad" />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('shows password quality when enabled and a password is entered', () => {
    const tree = renderer.create(<PasswordInput
      htmlId="test"
      name="test"
      onChange={() => {}}
      showQuality
      value="Uisi38#8iad" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

// Interaction test

