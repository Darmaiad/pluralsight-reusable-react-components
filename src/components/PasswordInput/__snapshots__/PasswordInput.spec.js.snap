// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hides password quality bar by default 1`] = `
<div
  style={
    Object {
      "marginBottom": 16,
    }
  }
>
  <label
    htmlFor="test"
    style={
      Object {
        "display": "block",
      }
    }
  >
    Password
     
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
       *
    </span>
  </label>
  <input
    id="test"
    maxLength={50}
    name="test"
    onChange={[Function]}
    placeholder={undefined}
    style={undefined}
    type="password"
    value="Uisi38#8iad"
  />
</div>
`;

exports[`shows password quality when enabled and a password is entered 1`] = `
<div
  style={
    Object {
      "marginBottom": 16,
    }
  }
>
  <label
    htmlFor="test"
    style={
      Object {
        "display": "block",
      }
    }
  >
    Password
     
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
       *
    </span>
  </label>
  <input
    id="test"
    maxLength={50}
    name="test"
    onChange={[Function]}
    placeholder={undefined}
    showQuality={true}
    style={undefined}
    type="password"
    value="Uisi38#8iad"
  />
</div>
`;
